Cairo Desktop/Cairo Desktop/TaskButton.xaml.cs

```diff
@@ -1,12 +1,41 @@
-﻿using CairoDesktop.Interop;
+﻿using CairoDesktop.AppGrabber;
+using CairoDesktop.Interop;
 using System;
 using System.Windows;
+using System.Windows.Controls;
 
 namespace CairoDesktop
 {
-	public partial class TaskButton
-	{
-        public WindowsTasks.ApplicationWindow Window;
+    public partial class TaskButton
+    {
+
+        private WindowsTasks.ApplicationWindow _applicationWindow;
+        public WindowsTasks.ApplicationWindow Window
+        {
+            get => _applicationWindow;
+            set
+            {
+                _applicationWindow = value;
+                InitializeApplicationInfo();
+            }
+        }
+
+        private bool hasInitializedApplicationInfoYet = false;
+        private ApplicationInfo _applicationInfo;
+        public ApplicationInfo App
+        {
+            get
+            {
+                if (!hasInitializedApplicationInfoYet)
+                    InitializeApplicationInfo();
```

Comment:
Any way we can speed this up? It seems pretty slow currently.

Reply:
Ill investigate.

Reply:
this is going to take some brain storming and effort. We can either comment it out or leave it in. Either way we need to find an efficient way of associating a process to an AppGrabber entry

Reply:
I'll take a stab at it, I have a couple ideas. We already do this somewhat with categories in the taskbar, but it needs to go a bit further in some cases.